web/versions.go

```diff
@@ -191,25 +191,37 @@ func getVersionScreenshot(c echo.Context) error {
 }
 
 func getVersionTarball(c echo.Context) error {
-	return getVersionAttachment(c, c.Param("tarball"))
-}
-
-func getVersionAttachment(c echo.Context, filename string) error {
-	appSlug := c.Param("app")
+	virtualSpace, space, err := getVirtualSpace(c)
+	if err != nil {
+		return err
+	}
+	slug := c.Param("app")
 	version := c.Param("version")
+	ver, err := registry.FindVersion(space, slug, version)
+	if err != nil {
+		return err
+	}
+	if ver == nil {
```

Comment:
We still have a function that can `return nil, nil`? We should modify it to return an error when the version is not found (or at least at boolean `ok` to say if the version was found).

Reply:
After verification, `FindVersion` return `nil, ErrVersionNotFound` if version not found.
I fix the code.